bpo-37689: add Path.is_relative_to() method (#14982)
python/cpython: Doc/library/pathlib.rst

      True

> .. method:: PurePath.is_relative_to(*other)

Reviewer: I think to name this parameter as `other_path` would be more explicit.

Author: Some other functions use `*other` too, such as: PurePath.relative_to(*other), PurePath.joinpath(*other). So I think keep consistency is ok. If most people like use `other_path`, i would update it.

Reviewer: Yes, I just checked `PurePath.relative_to` & `PurePath.joinpath`. In that case, it seems we're good on this part.

   Return whether or not this path is relative to the *other* path.

>       >>> from pathlib import PurePath

Reviewer 2: Nit: can you remove the import? Other snippets don't show the import.

Reviewer 3: What are the general guidelines for imports in interpreter examples? In the `pathlib` documentation, does anything from the `pathlib` module not require an import?

Reviewer 2: I don't know if there are official guidelines (you may ask on the ML). Personally, I think imports in interpreter snippets are most of the time a distraction, especially when there are tens of snippets on a single documentation page.

Generally speaking, I expect the user to be able to make out that the `PurePath` class listed in the documentation for the `pathlib` module needs the corresponding import before being used.

Reviewer 3: >Generally speaking, I expect the user to be able to make out that the PurePath class listed in the documentation for the pathlib module needs the corresponding import before being used.

@pitrou Thanks for the clarification, I think that's perfectly reasonable and agree that an excessive number of imports can be distracting. Unless someone else disagrees with your rationale, I don't think that a new ML thread is needed.

I think that the explicit imports are primarily appropriate when importing modules that are external to the documentation file. This can be see in one of the existing examples already in ``pathlib.rst``, when ``os`` is imported:
```
>>> import os
>>> p = PurePath('/etc')
>>> os.fspath(p)
'/etc'
```

Reviewer 2: > I think that the explicit imports are primarily appropriate when importing modules that are external to the documentation file.

Agreed.

Author: guys, thanks :)

      ValueError: '/etc/passwd' does not start with '/usr'

> .. method:: PurePath.is_relative_to(*other)

Reviewer 4: Please add a `versionadded` tag.

Author: thanks for review, done :)

Reviewer 2: Can you keep the list of methods in alphabetical order?

Author: of course.

>     Compute the boolean result that this path is relative to the path

Reviewer 3: Slight phrasing adjustment:
```suggestion
    Return the boolean result that this path is relative to *other* path.
```

Author: thanks for review, done:)

>     represented by *other* or not.

Reviewer 3: In combination with the above recommendation
```suggestion
```

>       >>> p = PurePosixPath('/etc/passwd')

Reviewer 3: I would suggest changing `PurePosixPath` to `PurePath`, `PurePath` will use `PurePosixPath` or `PureWindowsPath` based on the environment (for windows compatibility), and adding the import statement so that users can directly run the interpreter example.
```suggestion
      >>> from pathlib import PurePath
      >>> p = PurePath('/etc/passwd')
```

Author: you are right, show the PurePath demo would be better to user.

      ValueError: '/etc/passwd' does not start with '/usr'

.. method:: PurePath.is_relative_to(*other)

>    Return the boolean result that this path is relative to *other* path.

Reviewer 2: How about `Return whether this path is relative to the *other* path`?

Reviewer 3: I agree, this would be an improvement over my earlier suggestion. Since it's modifying a verb, "Return" in this case, I think it might be slightly more correct to explicitly specify the "or not":

``Return whether or not this path is relative to the *other* path``.

Author: ops， i am not a native english speaker, so i have no idea about "whether(or not)" :(

Reviewer 3: @shihai1991 That's okay, the distinction is often a pain, even for those of us who are native speakers (:
